src/types.c

```diff
@@ -225,7 +225,7 @@ reset_netdef(NetplanNetDefinition* netdef, NetplanDefType new_type, NetplanBacke
 
     netdef->optional = FALSE;
     netdef->optional_addresses = 0;
-    FREE_AND_NULLIFY(netdef->keep_configuration);
+    netdef->critical = NETPLAN_CRITICAL_FALSE;
```

Comment:
We probably shouldn't initialize `critical` with false.

According to the documentation, KeepConfiguration `defaults to "dhcp-on-stop" when systemd-networkd is running in initrd, "yes" when the root filesystem is a network filesystem, and "no" otherwise.`

Initializing it to false would imply setting it to false even when the default is something else. I suggest adding a new entry in the enum, something like `NETPLAN_CRITICAL_UNSET`, and use it as the default value. Then in the switch-case you added, you'd also have an option for the false case and and an option that does nothing for the unset case.

Reply:
+1 for using `NETPLAN_CRITICAL_UNSET` here.